src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -102,22 +106,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
   }
 
   private String getLink(String id, String query) {
-    String cse_request = CSE_URL + "?key=" + API_KEY + "&cx=" + id + "&q=" + query;
-    HttpRequest linkRequest =
-        HttpRequest.newBuilder()
-            .GET()
-            .uri(URI.create(cse_request))
-            .setHeader("User-Agent", "Java 11 HttpClient Bot")
-            .build();
-    HttpResponse<String> linkResponse;
-    try {
-      linkResponse = httpClient.send(linkRequest, HttpResponse.BodyHandlers.ofString());
-    } catch (IOException | InterruptedException ex) {
+    String url = CSE_URL + "?key=" + API_KEY + "&cx=" + id + "&q=" + query;
+    try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+      try (CloseableHttpResponse response = httpClient.execute(new HttpGet(url))) {
+
+        if (response.getStatusLine().getStatusCode() != 200) {
+          return null;
+        }
+        HttpEntity entity = response.getEntity();
+        if (entity == null) {
+          return null;
+        }
+        JsonParser parser = new JsonParser();
+        JsonObject obj;
+        try {
+          obj = parser.parse(new InputStreamReader(entity.getContent())).getAsJsonObject();
+        } catch (JsonIOException
```

Comment:
parser.parse is deprecated in 2.8.6 (version being used). Use JsonParser.parseReader.

This static method can also throw JsonParseException.

getAsJsonObject() can also throw IllegalStateException.

Reply:
There's a HTTP ERROR 500 java.lang.NoSuchMethodError: com.google.gson.JsonParser.parseReader(Lcom/google/gson/stream/JsonReader;)Lcom/google/gson/JsonElement; for using JsonParser.parseReader in 2.8.6

Reply:
Switching to 2.8.5 has no issues with being deprecated with the original code. JsonParser.parseReader doesn't exist in 2.8.5 and seems like 2.8.6 can't detect it at run time?

Reply:
I think it's working for me? If I switch to 2.8.6 and then modify the line to be:
`obj = JsonParser.parseReader(new InputStreamReader(entity.getContent())).getAsJsonObject();`, I don't see a 500 error. I might be missing something though, is that what you tried?

Reply:
Yeah, I used the exact same line, and I get the 500 error :/